### ISSUE TITLE
minifier: Syntax minification breaks the Temml library

### ISSUE THREAD

**Issue by cyyynthia** [oxc-project/oxc#24020]
While attempting to use [Temml](https://temml.org) in a project, the compiler completely bricks the library in several places, with chunks of code disappearing from the build entirely.

While I've initially observed this behaviour on a Rolldown-based bundle, after further investigation I've also found that some of the oxlint rules also lead to a broken output that does not function, which leads me to think there are (likely several) bad transformations and improper optimisations that discard important chunks of code.

Sorry, this report is quite a bit messy and may actually be surfacing *several* issues that emerge from processing `temml`... but I'm pretty confident the reproduction I've provided below should at least help triage and be a starting point for investigating 😅 

### Reproduction

The main reproduction for this issue should be the HTML snippet listed below, that can be bundled with Rolldown / Vite to see the problem in action.

In the oxc playground, the issue can be reproduced by copy/pasting the [`dist/temml.mjs` file from the temml package](https://npmx.dev/package-code/temml/v/0.13.3/dist%2Ftemml.mjs) into the playground. Enabling *Minify syntax* breaks the output, and that can be verified by copy/pasting the output directly into the HTML snippet instead of the import and opening it in your favourite Internet browser.

For reproducing the issue on oxlint, I didn't really investigate much but running `oxlint --type-aware --fix --fix-suggestion` with at least `restriction`, `style` and `pedantic` enabled on `dist/temml.mjs` yields a script broken as well (but with different symptoms).

<details>
<summary>reproduction.html</summary>

```html
<!DOCTYPE html>
<html>
	<title>temml breakage reproduction</title>
</html>
<body>
	<h1>Hello there</h1>
	<noscript>
		<p>Enable JS you fucking moron</p>
	</noscript>
	<p id="content" style="display: none">
		\def\d{\mathrm{d}}

		\oint_C \vec{B}\circ \d\vec{l} = \mu_0 \left( I_{\text{enc}} + \varepsilon_0 \frac{\d}{\d t} \int_S {\vec{E} \circ \hat{n}}\; \d a \right)

		\raisebox{0pt}{\Large%
		\textbf{Aaaa\raisebox{-0.3ex}{a}%
		\raisebox{-0.7ex}{aa}\raisebox{-1.2ex}{r}%
		\raisebox{-2.2ex}{g}\raisebox{-4.5ex}{h}}}

		\ce{Zn^2+ &lt;=&gt;[+ 2OH-][+ 2H+]
		$\underset{\text{amphoteres Hydroxid}}
		{\ce{Zn(OH)2 v}}$ &lt;=&gt;[+ 2OH-][+ 2H+]
		$\underset{\text{Hydroxozikat}}
		{\ce{[Zn(OH)4]^2-}}$}
	</p>
	<script type="module">
		import temml from 'temml'

		const el = document.getElementById('content')

		const lines = el.innerText.split('\n')
		const f = lines.find((s) => s.trim())
		const whitespace = f.length - f.trimStart().length
		const latex = lines.map((l) => l.substring(whitespace)).join('\n')

		el.innerText = ''
		el.style.display = ''
		temml.render(latex, el)
	</script>
</body>
```

</details>

<details>
<summary>Expected v. Actual</summary>

<img width="784" height="193" alt="Image" src="https://github.com/user-attachments/assets/b0caa219-451d-4ebf-8dea-b26929a48d10" />

<img width="1043" height="486" alt="Image" src="https://github.com/user-attachments/assets/1cbd2895-a8c6-466a-96e5-187d1db21d86" />

</details>

**Comment by camc314** [comment]
Hi! Thanks for reporting this issue.

This seems to have the same root cause of https://github.com/oxc-project/oxc/issues/15524 & #16886.

Inside the package, there is the following code:

<img width="920" height="478" alt="Image" src="https://github.com/user-attachments/assets/cee0867a-4215-4d53-b0dd-6c4437b70cdd" />

This is optimized by the minifier into a single template string which is incorrectly printed.

[Playground link](https://playground.oxc.rs/?options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Afalse%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22compress%22%3Atrue%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22tsx%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%7D%2C%22formatter%22%3A%7B%22useTabs%22%3Afalse%2C%22tabWidth%22%3A2%2C%22endOfLine%22%3A%22lf%22%2C%22printWidth%22%3A80%2C%22singleQuote%22%3Afalse%2C%22jsxSingleQuote%22%3Afalse%2C%22quoteProps%22%3A%22as-needed%22%2C%22trailingComma%22%3A%22all%22%2C%22semi%22%3Atrue%2C%22arrowParens%22%3A%22always%22%2C%22bracketSpacing%22%3Atrue%2C%22bracketSameLine%22%3Afalse%2C%22objectWrap%22%3A%22preserve%22%2C%22singleAttributePerLine%22%3Afalse%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%2C%22optimizeEnums%22%3Atrue%2C%22optimizeConstEnums%22%3Atrue%7D%2C%22isolatedDeclarations%22%3A%7B%22stripInternal%22%3Afalse%7D%2C%22codegen%22%3A%7B%22normal%22%3Atrue%2C%22jsdoc%22%3Atrue%2C%22annotation%22%3Atrue%2C%22legal%22%3Atrue%7D%2C%22compress%22%3A%7B%7D%2C%22mangle%22%3A%7B%22topLevel%22%3Atrue%2C%22keepNames%22%3Afalse%7D%2C%22controlFlow%22%3A%7B%22verbose%22%3Afalse%7D%2C%22inject%22%3A%7B%22inject%22%3A%7B%7D%7D%2C%22define%22%3A%7B%22define%22%3A%7B%7D%7D%7D&code=const+spaceRegexString+%3D+%22%5B+%5Cr%5Cn%5Ct%5D%22%3B%0Aconst+controlWordRegexString+%3D+%22%5C%5C%5C%5C%5Ba-zA-Z%40%5D%2B%22%3B%0Aconst+controlSymbolRegexString+%3D+%22%5C%5C%5C%5C%5B%5E%5CuD800-%5CuDFFF%5D%22%3B%0Aconst+controlWordWhitespaceRegexString+%3D+%60%28%24%7BcontrolWordRegexString%7D%29%24%7BspaceRegexString%7D*%60%3B%0Aconst+controlSpaceRegexString+%3D+%22%5C%5C%5C%5C%28%5Cn%7C%5B+%5Cr%5Ct%5D%2B%5Cn%3F%29%5B+%5Cr%5Ct%5D*%22%3B%0Aconst+combiningDiacriticalMarkString+%3D+%22%5B%5Cu0300-%5Cu036f%5D%22%3B%0Aconst+combiningDiacriticalMarksEndRegex+%3D+new+RegExp%28%60%24%7BcombiningDiacriticalMarkString%7D%2B%24%60%29%3B%0Aconst+tokenRegexString+%3D%0A++%60%28%24%7BspaceRegexString%7D%2B%29%7C%60+%2B+%2F%2F+whitespace%0A++%60%24%7BcontrolSpaceRegexString%7D%7C%60+%2B++%2F%2F+whitespace%0A++%22%28%5B%21-%5C%5C%5B%5C%5C%5D-%5Cu2027%5Cu202A-%5CuD7FF%5CuF900-%5CuFFFF%5D%22+%2B+%2F%2F+single+codepoint%0A++%60%24%7BcombiningDiacriticalMarkString%7D*%60+%2B+%2F%2F+...plus+accents%0A++%22%7C%5B%5CuD800-%5CuDBFF%5D%5B%5CuDC00-%5CuDFFF%5D%22+%2B+%2F%2F+surrogate+pair%0A++%60%24%7BcombiningDiacriticalMarkString%7D*%60+%2B+%2F%2F+...plus+accents%0A++%22%7C%5C%5C%5C%5Cverb%5C%5C*%28%5B%5E%5D%29.*%3F%5C%5C4%22+%2B+%2F%2F+%5Cverb*%0A++%22%7C%5C%5C%5C%5Cverb%28%5B%5E*a-zA-Z%5D%29.*%3F%5C%5C5%22+%2B+%2F%2F+%5Cverb+unstarred%0A++%60%7C%24%7BcontrolWordWhitespaceRegexString%7D%60+%2B+%2F%2F+%5CmacroName+%2B+spaces%0A++%60%7C%24%7BcontrolSymbolRegexString%7D%29%60%3B+%2F%2F+%5C%5C%2C+%5C%27%2C+etc.)

@Dunqing handing back to you 🙂

**Comment by Dunqing** [comment]
Fixed by #25084. Verified temml@0.13.3 renders identically after compress + mangle on current main.
